h11/tests/test_events.py

```diff
@@ -94,6 +94,13 @@ def test_events():
             headers=[("Host", "a"), ("Foo", "asd\x01\x02\x7f")],
             http_version="1.0")
 
+    # Request target is validated
+    for bad_char in b"\x20\x7f\xee":
```

Comment:
Can you add `\x00` to the list of test characters? That's the one most likely to mess people up if we let it through.